src/main/scala/org/broadinstitute/dsde/workbench/sam/service/KeyCache.scala

```diff
@@ -0,0 +1,23 @@
+package org.broadinstitute.dsde.workbench.sam.service
+
+import org.broadinstitute.dsde.workbench.model.google.{GoogleProject, ServiceAccountKey, ServiceAccountKeyId, ServiceAccountPrivateKeyData}
+import org.broadinstitute.dsde.workbench.model.{WorkbenchUser, WorkbenchUserId}
+
+import scala.concurrent.Future
+
+/**
+  * Created by mbemis on 1/10/18.
+  */
+trait KeyCache {
+  def onBoot(): Future[Unit]
+  def getKey(user: WorkbenchUser, project: GoogleProject): Future[String]
```

Comment:
what if the parameter here is the PetServiceAccount? It has all the information and then the cache code can rely on the fact that the pet already exists.